server/usecase/team_test.go

```diff
@@ -0,0 +1,134 @@
+package usecase_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/traPtitech/piscon-portal-v2/server/domain"
+	"github.com/traPtitech/piscon-portal-v2/server/repository/mock"
+	"github.com/traPtitech/piscon-portal-v2/server/usecase"
+	"go.uber.org/mock/gomock"
+)
+
+func TestCreateTeam(t *testing.T) {
+	t.Parallel()
+
+	ctrl := gomock.NewController(t)
+
+	mockRepo := mock.NewMockRepository(ctrl)
+	teamUseCase := usecase.NewTeamUseCase(mockRepo)
+
+	mockRepo.EXPECT().FindUser(gomock.Any(), gomock.Any()).
+		DoAndReturn(func(_ context.Context, id string) (domain.User, error) {
+			return domain.User{ID: id}, nil
+		}).
+		AnyTimes()
+	mockRepo.EXPECT().CreateTeam(gomock.Any(), gomock.Any()).AnyTimes()
+
+	userID := uuid.NewString()
+
+	tests := []struct {
+		name    string
+		input   usecase.CreateTeamInput
+		wantErr bool
+	}{
+		{
+			name: "valid",
+			input: usecase.CreateTeamInput{
+				Name:      "valid-test-team",
+				MemberIDs: []string{userID},
+				CreatorID: userID,
+			},
+			wantErr: false,
+		},
+		{
+			name: "multiple members",
+			input: usecase.CreateTeamInput{
+				Name:      "multiple-members-test-team",
+				MemberIDs: []string{userID, uuid.NewString()},
+				CreatorID: userID,
+			},
+			wantErr: false,
+		},
+		{
+			name: "4 members",
+			input: usecase.CreateTeamInput{
+				Name:      "4-members-test-team",
+				MemberIDs: []string{userID, uuid.NewString(), uuid.NewString(), uuid.NewString()},
+				CreatorID: userID,
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := teamUseCase.CreateTeam(context.TODO(), tt.input)
```

Comment:
`context.TODO()`よりは`context.Background()`な気がする。

> Code should use context.TODO when it's unclear which Context to use or it is not yet available (because the surrounding function has not yet been extended to accept a Context parameter).

https://pkg.go.dev/context#TODO